File: javascript/cluster_management/src/create-single-region.js
import { DSQLClient } from "@aws-sdk/client-dsql";
import { CreateClusterCommand } from "@aws-sdk/client-dsql";

async function createCluster(client, tags, deletionProtectionEnabled) {
    const createClusterCommand = new CreateClusterCommand({
        deletionProtectionEnabled: deletionProtectionEnabled,
        tags,
    });
    try {
        const response = await client.send(createClusterCommand);
        return response;
    } catch (error) {
        console.error("Failed to create cluster: ", error.message);
    }
}

async function main() {
    const region = "us-east-1";
    const client = new DSQLClient({ region });
    const tags = { Name: "FooBar"};
    const deletionProtectionEnabled = true;

    const response = await createCluster(client, tags, deletionProtectionEnabled);
    console.log("Cluster Id:", response.identifier);
}

main();

File: ruby/cluster_management/lib/create_single_region.rb
require 'aws-sdk-core'
require 'aws-sdk-dsql'

def create_cluster(region)
  begin
    # Create client with default configuration and credentials
    client = Aws::DSQL::Client.new(region: region)

    response = client.create_cluster(
      deletion_protection_enabled: true,
      tags: {
        "Name" => "example_cluster_ruby"
        }
    )

    # Extract and verify response data
    identifier = response.identifier
    arn = response.arn
    puts arn
    raise "Unexpected status when creating cluster: #{response.status}" unless response.status == 'CREATING'
    raise "Deletion protection not enabled" unless response.deletion_protection_enabled

    [identifier, arn]
  rescue Aws::Errors::ServiceError => e
    raise "Failed to create cluster: #{e.message}"
  end
end
